test/util/statsd.test.ts

```diff
@@ -0,0 +1,168 @@
+import * as assert from "power-assert";
+import { defaultConfiguration } from "../../lib/configuration";
+import { StatsdAutomationEventListener } from "../../lib/util/statsd";
+import {
+    defaultStatsdAdapterConfig,
+    StatsdAdapter,
+    StatsdAdapterCallback,
+    StatsdAdapterOptions,
+    StatsdAdapterStat,
+    StatsdAdapterTags,
+} from "../../lib/util/statsdAdapter";
+
+import * as HotShots from "hot-shots";
+import * as NodeStatsD from "node-statsd";
+import * as StatsdClient from "statsd-client";
+
+describe("StatsdAutomationEventListener", () => {
+    describe("constructor", () => {
+        it("works with default configuration", () => {
+            const config = defaultConfiguration();
+            assert.doesNotThrow(() => {
+                const statsdListener = new StatsdAutomationEventListener(config);
+            });
+        });
+        it("works with hot-shots statsd", () => {
+            const config = defaultConfiguration();
+            const expectedStatsdAdapterConfig = defaultStatsdAdapterConfig(config);
+            config.statsd.adapterConfig = {
+                adaptClient: (adapterOptions: StatsdAdapterOptions) => {
+                    assert.deepEqual(adapterOptions, expectedStatsdAdapterConfig.adapterOptions);
+
+                    return new HotShots.StatsD(adapterOptions);
+                },
+            };
+            assert.doesNotThrow(() => {
+                const statsdListener = new StatsdAutomationEventListener(config);
+            });
+        });
+
+        it("works with node-statsd", () => {
+            const config = defaultConfiguration();
+            const expectedStatsdAdapterConfig = defaultStatsdAdapterConfig(config);
+
+            class NodeStatsDAdapter implements StatsdAdapter {
```

Comment:
Can we make that a real class in the lib dir and make it the default? If people use the DataDog extensions they can configure the hot-shots based implementation.

Reply:
That would be a breaking change in case that people are already using a datadog enabled statsd server, no? They might suddenly stop receiving the datadog events and not realize it.

Reply:
That's true. Let's keep the current default then.